Framework/DataHandling/src/SaveReflectometryAscii.cpp

```diff
@@ -187,13 +186,26 @@ std::string SaveReflectometryAscii::sampleInfo(const std::string &logName) {
   return "Not defined";
 }
 
+/// Retrieve sample log unit
+std::string SaveReflectometryAscii::sampleUnit(const std::string &logName) {
+  auto run = m_ws->run();
+  try {
+    return " " +
+           boost::lexical_cast<std::string>(run.getLogData(logName)->units());
+  } catch (Exception::NotFoundError &) {
+  }
+  return "";
+}
+
 /// Write one header line
 void SaveReflectometryAscii::writeInfo(const std::string logName,
                                        const std::string logValue) {
   if (!logValue.empty())
-    m_file << logName << " : " << sampleInfo(logValue) << '\n';
+    m_file << logName << " : " << sampleInfo(logValue) << sampleUnit(logValue)
```

Comment:
shouldn't the second part be `sampleUnit(logName)`? An improvement is to declare a constant at the top of the function `const std::string logUnit = sampleUnit(logName);` and use that everywhere below.

Reply:
Shouldn't the line be
```c++
m_file << logName << " : " << sampleInfo(logValue) << sampleUnit(logName)
```